src/EFCore.Design/Migrations/Design/MigrationsScaffolder.cs

```
@@ -72,23 +78,31 @@ public MigrationsScaffolder([NotNull] MigrationsScaffolderDependencies dependenc
             string language = null)
         {
             Check.NotEmpty(migrationName, nameof(migrationName));
-            Check.NotEmpty(rootNamespace, nameof(rootNamespace));
+            Check.NotEmpty(string.Empty + rootNamespace + subNamespace, "rootNamespace + subNamespace");
```

😬 There's a lot I don't like about this line. That expression is ...odd, and it will end up throwing ArgumentException with a ParamName of "rootNamespace + subNamespace".

It's probably not worth validating this. What happens if we allow it? We generate invalid C#? (like #2467)

---

I like to fail early :-) But that said, I'll take it out - `CSharpMigrationsGenerator` checks that the namespace is `NotEmpty` in all three of `GenerateMigration`, `GenerateMetadata` and `GenerateSnapshot` - so it would just fail there instead.